backend/src/apiserver/list/list.go

```diff
@@ -290,6 +290,7 @@ func (o *Options) nextPageToken(listable Listable) (*token, error) {
 		KeyFieldName:     listable.PrimaryKeyColumnName(),
 		KeyFieldValue:    keyField.Interface(),
 		IsDesc:           o.IsDesc,
+		Filter:           o.Filter,
```

Comment:
can we verify the token can be correctly marshalled into expected string?

Reply:
Great catch @IronPan ! This was completely missing from before. Best part is, we had a test that would have caught it in experiment_store_test.go, but somehow, it too was testing the wrong value! Fixed this and added tests for serialization/deserialization of token and filter. PTAL.